### PR TITLE
Align ERC721 Receiver with current ERC721 standard.

### PR DESCRIPTION
**Pull request by paulbarclay**
# 🚀 Description

The ERC-721 onERC721Received function changed. It's now `onERC721Received(address,address,uint256,bytes)`. This makes the function signature `0x150b7a02` (it was 0xf0b9e5ba)

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

File: contracts/token/ERC721/ERC721Receiver.sol [hunk 1]
```diff
@@ -21,12 +21,14 @@ contract ERC721Receiver {
    * transfer. This function MUST use 50,000 gas or less. Return of other
    * than the magic value MUST result in the transaction being reverted.
    * Note: the contract address is always the message sender.
-   * @param _from The sending address
+   * @param _operator The address which called `safeTransferFrom` function
+   * @param _from The address which previously owned the token
    * @param _tokenId The NFT identifier which is being transfered
    * @param _data Additional data with no specified format
    * @return `bytes4(keccak256("onERC721Received(address,uint256,bytes)"))`
```
Comment (shrugs) [inline contracts/token/ERC721/ERC721Receiver.sol]: please update the text here as well

File: contracts/token/ERC721/ERC721Holder.sol [hunk 2]
```diff
@@ -4,7 +4,7 @@ import "./ERC721Receiver.sol";
 
 
 contract ERC721Holder is ERC721Receiver {
-  function onERC721Received(address, uint256, bytes) public returns(bytes4) {
+  function onERC721Received(address, address, uint256, bytes) public returns(bytes4) {
```
Comment (shrugs) [inline contracts/token/ERC721/ERC721Holder.sol]: Q: What is the purpose of this file? would ERC721Receiver not be the more expressive implementation?

File: contracts/token/ERC721/ERC721Receiver.sol (hunk 1 above)
Comment (paulbarclay) [reply contracts/token/ERC721/ERC721Receiver.sol]: Done. Also fixed the "This function MUST use 50,000 gas or less." statement, which is no longer true.

File: contracts/token/ERC721/ERC721Holder.sol (hunk 2 above)
Comment (paulbarclay) [reply contracts/token/ERC721/ERC721Holder.sol]: The "Receiver" file is the interface as defined in the standard; the "Holder" contract is the minimal implementation of that interface, so I think we have to leave this as is.

I'm not a huge fan of interfaces not being clearly named as such; I don't particularly like the "Holder" name here, but I don't have a deep enough view of the whole OpenZeppelin stack to guarantee that any change I make would actually be an improvement, so I'll leave this one to others.

Comment (shrugs) [reply contracts/token/ERC721/ERC721Holder.sol]: Ah, I realize that now, yeah. I do think we'd benefit from an `I` prefix being applied to all of the interfaces, but that's a larger change. Cheers

**Comment by johnhforrest** [comment]
#1032 is safe to close after this gets merged.

**Pull request merged**